src/page-modules/assistant/layout.tsx

```diff
@@ -82,6 +86,11 @@ function AssistantLayout({ children, tripQuery }: AssistantLayoutProps) {
     setIsSwapping(false);
   };
 
+  const onClearViaLocation = () => {
+    tripQuery.via = null;
```

Comment:
Instead of modifying the object and setting `via` to `null`, I would probably pass a new object that spreads `tripQuery` and sets `via: null`. 

Not sure if modifying the object in this case results in any side effects, 


```ts
  const onClearViaLocation = () => {
    setValuesWithLoading({ ...tripQuery, via: null });
  };
```

Reply:
I will check it out👍